### PR TITLE
Gh-3299: Add ChangeGraphId operation to POC simple federated store

### LINKED ISSUE #3299: Add ChangeGraphId operation to simple federated POC
Add in the ChangeGraphId operation to the new simple federated POC. This is mostly important for accumulo and renaming the tables.

### PR DESCRIPTION
**Pull request by cn337131**
PR adds a ChangeGraphId, mainly for accumulo stores, to the POC federated store. 

# Related issue

- Resolve #3299

### CODE REVIEW

File: store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java [hunk 1]
```diff
@@ -186,6 +194,59 @@ public void setDefaultGraphIds(final List<String> defaultGraphIds) {
...
+     *
+     */
+    private void updateTableAndCache(final GraphSerialisable graphToUpdate, final String newGraphId) throws OperationException {
+        final String graphToUpdateId = graphToUpdate.getGraphId();
+
+        // Remove from cache
+        removeGraph(graphToUpdateId);
+
+        // Update accumulo tables with new graph ID
+        if (graphToUpdate.getStoreProperties().getStoreClass().startsWith(AccumuloStore.class.getPackage().getName())) {
```
Comment (tb06904) [inline store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java:231]: Does this work on a map store? If not probably worth throwing exception or warning.

File: store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java [hunk 2]
```diff
@@ -186,6 +194,59 @@ public void setDefaultGraphIds(final List<String> defaultGraphIds) {
...
+            try {
+                renameTable((AccumuloProperties) graphToUpdate.getStoreProperties(), graphToUpdateId, newGraphId);
+            } catch (final StoreException e) {
+                throw new OperationException("Error trying to rename Accumulo table for graph: " + graphToUpdateId, e);
+            }
+        }
+
+        GraphSerialisable updatedGraphSerialisable = new GraphSerialisable.Builder(graphToUpdate)
```
Comment (tb06904) [inline store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java:239]: Can't you just do somthing like `graphToUpdate.getConfig().setGraphId(newGraphId)` then add back to the cache?

File: store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java (hunk 1 above)
Comment (cn337131) [reply store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java:231]: yeah it works on the map - it just removes it from the cache and re-adds it with a new id but skips the renaming tables bit

File: store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java [hunk 3]
```diff
@@ -186,6 +193,44 @@ public void setDefaultGraphIds(final List<String> defaultGraphIds) {
...
+    /**
+     * Change the graph's ID for the specified graph.
+     *
+     * @param graphToUpdateId the graph that is to have its ID updated
+     * @param newGraphId the new graph ID
+     * @throws OperationException if the operation fails
+     * @throws StoreException if the accumulo tables cannot be renamed
+     *
+     */
+    public void changeGraphId(final String graphToUpdateId, final String newGraphId) throws OperationException, StoreException {
```
[the remark this answers is not in the document]
Comment (cn337131) [reply store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java:205]: it issss

**Review by tb06904 — APPROVED** [review]

**Review by tb06904 — APPROVED** [review]

**Review by p29876 — APPROVED** [review]

**Pull request merged**